File: EVENPSUM.go
package main

import (
	"fmt"
)

func findOdd(n int) (odd int) {
	var isNOdd bool
	if n%2 == 0 {
		isNOdd = false
	} else {
		isNOdd = true
	}
	if isNOdd == true {
		odd = n/2 + 1
	} else {
		odd = n / 2
	}
	return odd
}
func main() {
	var testCases, A, B, counter int
	fmt.Scanf("%d", &testCases)
	for i := 0; i < testCases; i++ {
		counter = 0
		fmt.Scanf("%d %d", &A, &B)
		oddA := findOdd(A)
		oddB := findOdd(B)
		evenA := A - oddA
		evenB := B - oddB
		counter = (oddA * oddB) + (evenA * evenB)

		fmt.Println(counter)
	}
}

File: EVENPSUM.py
test_cases = int(input())


def oddnum(n):
    odd = False if n % 2 == 0 else False
    if odd:
        return n//2+1
    else:
        return n//2


for _ in range(test_cases):
    count = 0
    A, B = map(int, input().strip().split())
    oddA = oddnum(A)
    oddB = oddnum(B)
    count = (oddA*oddB)+((A-oddA)*(B-oddB))
    print(count)
